extension/observer/ecstaskobserver/extension.go

```diff
@@ -15,21 +15,102 @@
 package ecstaskobserver // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer/ecstaskobserver"
 
 import (
+	"context"
+	"fmt"
+	"strconv"
+
 	"go.opentelemetry.io/collector/component"
+	"go.uber.org/zap"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/ecsutil"
 )
 
+const runningStatus = "RUNNING"
+
 var _ component.Extension = (*ecsTaskObserver)(nil)
 var _ observer.EndpointsLister = (*ecsTaskObserver)(nil)
+var _ observer.Observable = (*ecsTaskObserver)(nil)
 
 type ecsTaskObserver struct {
 	component.Extension
+	*observer.EndpointsWatcher
 	config           *Config
-	endpointsWatcher *observer.EndpointsWatcher
+	metadataProvider ecsutil.MetadataProvider
 	telemetry        component.TelemetrySettings
 }
 
+func (e *ecsTaskObserver) Shutdown(ctx context.Context) error {
```

Comment:
Wondering if `Shutdown` should go towards the end of the file since it's done last?

Reply:
Since this is the `Extension`/`Component` method I put it first by convention.